test/featuretransforms.jl

```diff
@@ -0,0 +1,363 @@
+                        :price => KeyedArray(ones(3, 4); time=1:3, id=[:a, :b, :a, :b]),
+                    ]
+                ])...
+            )
+
+            r = FeatureTransforms.apply(ds, T, (:_, :price, :_))
+
+            @test r isa KeyedDataset
+            @test isequal(r, expected)
+            @test !isequal(ds, expected)
+        end
+
+        @testset "using symbol" begin
+            expected = KeyedDataset(
+                flatten([
+                    :train => [
+                        :load => KeyedArray(ones(3, 4); time=1:3, loc=[:x, :y, :x, :y]),
+                        :price => KeyedArray(M2; time=1:3, id=[:a, :b]),
+                    ],
+                    :predict => [
+                        :load => KeyedArray(ones(3, 4); time=1:3, loc=[:x, :y, :x, :y]),
+                        :price => KeyedArray(M4; time=1:3, id=[:a, :b]),
+                    ]
+                ])...
+            )
+
+            r = FeatureTransforms.apply(ds, T, :loc)
+
+            @test r isa KeyedDataset
+            @test isequal(r, expected)
+            @test !isequal(ds, expected)
+        end
+
+        @testset "using dims" begin
+            expected = KeyedDataset(
+                flatten([
+                    :train => [
+                        :load => KeyedArray(ones(3, 2); time=1:3, loc=[:x, :x]),
+                        :price => KeyedArray(M2; time=1:3, id=[:a, :b]),
+                    ],
+                    :predict => [
+                        :load => KeyedArray(ones(3, 2); time=1:3, loc=[:x, :x]),
+                        :price => KeyedArray(M4; time=1:3, id=[:a, :b]),
+                    ]
+                ])...
+            )
+
+            r = FeatureTransforms.apply(ds, T, :loc; dims=2, inds=[1])
+
+            @test r isa KeyedDataset
+            @test isequal(r, expected)
+            @test !isequal(ds, expected)
+        end
+    end
+
+    @testset "ManyToOne" begin
+        T = FakeManyToOneTransform()
+
+        @testset "default applies to all components" begin
+            expected = KeyedDataset(
+                # ideally we would drop the :time constraint when it gets reduced
+                OrderedSet(Pattern[(:__, :time), (:__, :loc), (:__, :id)]),
+                LittleDict(flatten([
+                    :train => [
+                        :load => KeyedArray(ones(2); loc=[:x, :y]),
+                        :price => KeyedArray(ones(2); id=[:a, :b]),
+                    ],
+                    :predict => [
+                        :load => KeyedArray(ones(2); loc=[:x, :y]),
+                        :price => KeyedArray(ones(2); id=[:a, :b]),
+                    ]
+                ])...)
+            )
+
+            r = FeatureTransforms.apply(ds, T; dims=:time)
+
+            @test r isa KeyedDataset
+            @test isequal(r, expected)
+            # https://github.com/mcabbott/AxisKeys.jl/issues/61
+            @test_broken !isequal(ds, expected)
```

Comment:
`isequal` is broken upstream

https://github.com/mcabbott/AxisKeys.jl/issues/61